src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -1250,6 +1250,70 @@ public static void CheckParseDepth()
             Assert.Throws<JsonReaderException>(() => JsonDocument.Parse(badJson));
         }
 
+        [Fact]
+        public static void PayloadWithTooLargeMaxDepthCapped()
+        {
+            // MaxDepthOverflow * 8 > int.MaxValue
+            const int MaxDepthOverflow = 1 << 28; //268_435_456;
+
+            string badJson = new string('[', MaxDepthOverflow) + "2" + new string(']', MaxDepthOverflow);
+
+            Assert.Throws<JsonReaderException>(() => JsonDocument.Parse(badJson, new JsonReaderOptions { MaxDepth = 7 }));
```

Comment:
Why is this writing a quarter million nested array if it's going to fail when it hits the 8th one?

Reply:
Since it isn't really testing something unique, I will remove it.

I was mainly trying to test that a high depth payload doesn't unexpectedly fail.

I was thinking of adding some outer loop tests which would work on deeply nested payloads, but they take way too long and may not be something we want to codify.
```C#
        [OuterLoop]
        [Fact]
        public static void PayloadWithTooLargeMaxDepth()
        {
            // MaxDepthOverflow * 8 > int.MaxValue
            const int MaxDepthOverflow = 1 << 28; //268_435_456;

            string badJson = new string('[', MaxDepthOverflow) + "2" + new string(']', MaxDepthOverflow);

            Assert.Throws<ArgumentOutOfRangeException>(() => JsonDocument.Parse(badJson, new JsonReaderOptions { MaxDepth = int.MaxValue }));
        }

        [OuterLoop]
        [Fact]
        public static void PayloadWithMaxSupportedDepth()
        {
            const int MaxAllowedDepth = 41_666_665;

            string okayJson = new string('[', MaxAllowedDepth) + "2" + new string(']', MaxAllowedDepth);

            using (JsonDocument doc = JsonDocument.Parse(okayJson, new JsonReaderOptions { MaxDepth = int.MaxValue }))
            {
            }
        }
```